kpops/pipeline_generator/pipeline.py

```diff
@@ -83,19 +155,34 @@ def validate_unique_names(self) -> None:
 
     def generate_graph(self) -> None:
         for component in self.components:
-            self.graph.add_node(component.id)
+            node_component = InternalNodeRepresentation(
+                name=component.id, component=component, is_topic=False
+            )
+            self._component_index[node_component.name] = node_component
```

Comment:
what is the component index for, isn't that very similar to the graph? do you just need it for the `is_topic` distinction?
(and why not use the `id` as index key)

Reply:
I can explain it more deeply in the meeting, but to be able to make pydantic models nodes, I would need to froze them and make it inmutable. I didn't wanted to take a risk to do something that could break things in many parts. So what I do is a little index that let me retrieve the component from the node name.

Reply:
btw we actually use id as index key it is just a rename of the `InternalNodeRepresentation` object